Draw decorative elements in the pads

henrebotha opened the pull request:
This adds a frivolous decorative element to add some peaceful \~waves\~ to the surrounding blank space.

Features:

- Optional (default off).
- Decorative string is customisable (some nice ones to try: `/`, `.`, `*`, `†`) and can be of arbitrary length.
- Rarity is customisable (higher number = fewer characters drawn).

<img width="953" height="796" alt="image" src="https://github.com/user-attachments/assets/02744ea9-3aae-4f21-99a2-d9685ed6ddaf" />

henrebotha committed:
Draw decorative elements in the pads

junegunn reviewed (commented):
Haha, pretty. If you're serious about getting this merged, could you update the code to remove the `l:` prefixes? I don't use them.

henrebotha commented:
Done.

junegunn commented:
Thanks,

It would be nice if it can randomly sprinkle strings of arbitrary lengths instead of just one fixed string.

```sh
let g:goyo_decoration_elements = ['⭐', '✨', '💫', '🌟', '🎅🤶🧑‍🎄']
```

Also, I think the current term "rarity" might be confusing for users. If I understood correctly, decorations are printed with a `1 / rarity` probabilty. But I think it would be more straighforward to rename it to "density" that should be within zero to one range, with the default value of zero. That way, we can remove `g:goyo_decorate`, and users can simply enable decorations with something like:

```vim
let g:goyo_decoration_density = 0.05
```

henrebotha commented:
> It would be nice if it can randomly sprinkle strings of arbitrary lengths instead of just one fixed string.

Yeah I think that's what I originally had in mind (I liked the idea of variable-length waves, `~` or `~~` or `~~~`), but I couldn't quite get it working right and then probably got distracted. I'll take another crack at it.

> Also, I think the current term "rarity" might be confusing for users. If I understood correctly, decorations are printed with a `1 / rarity` probabilty. But I think it would be more straighforward to rename it to "density" that should be within zero to one range, with the default value of zero. That way, we can remove `g:goyo_decorate`

I agree "density" is more intuitive. I'll make the change.

henrebotha commented:
Both changes made.

henrebotha commented:
Whoops, `l:` prefixes snuck back in. Fixed.

junegunn commented on doc/goyo.txt:
```diff
@@ -82,10 +82,13 @@ CONFIGURATION                                               *goyo-configuration*
 ==============================================================================
 
                                       *g:goyo_width* *g:goyo_height* *g:goyo_linenr*
+                                *g:goyo_decoration_chars* *g:goyo_decoration_rarity*
```
We need to update this line as well.

junegunn committed:
Change default value of g:goyo_decoration_elements

Elements with mixed lengths can produce an unintuitive result when the
density is set to 1.

junegunn committed:
Handle empty element

junegunn committed:
Handle density smaller than 0.01

junegunn committed:
Reformat vimdoc

junegunn commented:
Merged, thanks!
